sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1450,6 +1450,26 @@ kernel_id ProgramManager::getSYCLKernelID(const std::string &KernelName) {
   return KernelID->second;
 }
 
+bool ProgramManager::hasCompatibleImage(const device &Dev) {
```

Comment:
maybe instead of running two loops and creating a temporary set, would it be possible to use `std::find` or `std::any_of` on the DeviceImages collection ? That might be easier to read and quicker.

Reply:
Updated with a slightly easier to use map and use of `std::any_of`